specs/public-api.yaml

```diff
@@ -68,15 +68,80 @@ paths:
         - Decision
       security:
         - ApiKeyAuth: []
-      description: List existing decisions for the organization.
-      summary: Query a list of decisions
+      description: List matching decisions for the organization
+      summary: List decisions based on the provided filters and pagination
+      parameters:
+        - name: outcome[]
+          description: decision outcomes used to filter the list
+          in: query
+          required: false
+          schema:
+            type: array
+            items:
+              $ref: "#/components/schemas/outcome"
+        - name: scenario_id[]
+          description: scenario IDs used to filter the list
+          in: query
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+        - name: trigger_object[]
+          description: trigger objects used to filter the list
+          in: query
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+        - $ref: "#/components/parameters/start_date"
+        - $ref: "#/components/parameters/end_date"
+        - name: has_case
+          description: the decision is attached to a case
+          in: query
+          required: false
+          schema:
+            type: boolean
+        - name: scheduled_execution_id[]
+          description: scheduled execution IDs used to filter the list
+          in: query
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+        - name: sorting
+          description: the field used to sort the items
```

Comment:
could you please double check what happens if you pass this param with a value other than created_at ?

Reply:
Error 500 is returned
```JSON
{
    "message": "An unexpected error occurred. Please try again later, or contact support if the problem persists.",
    "error_code": ""
}
```

There is a check in the repository to ensure we respect the whitelist
```go
	var offsetField any
	switch p.Sorting {
	case models.DecisionSortingCreatedAt:
		offsetField = offset.CreatedAt
	default:
		// only pagination by created_at is allowed for now
		return query, fmt.Errorf("invalid sorting field: %w", models.BadParameterError)
	}
```

Reply:
Ok for me, but I will handle this in a separated PR.

I need to merge this to match the merged PR in the front end

Reply:
yep sure 👍